publishing-sdk/src/main/java/com/ably/tracking/publisher/workerqueue/EventWorkerQueue.kt

```diff
@@ -44,6 +46,13 @@ internal class EventWorkerQueue(
     private fun handleWorkResult(workResult: WorkResult) {
         val resultHandler = getWorkResultHandler(workResult)
         val nextWorker = resultHandler.handle(workResult, corePublisher)
-        nextWorker?.let { corePublisher.request(it.event) }
+        nextWorker?.let {
+            it.event.apply {
+                when (this) {
+                    is Request<*> -> corePublisher.request(this)
+                    is AdhocEvent -> corePublisher.enqueue(this)
+                }
+            }
```

Comment:
I would choose some other scope function for this use-case, as the `apply` does not seem very idiomatic here :thinking: According to the [docs](https://kotlinlang.org/docs/scope-functions.html#apply) the `apply` function is mostly used for object configuration and here we're doing something different :wink: 

I'd either use the `let` function but with a named parameter, something like this
```suggestion
            it.event.let { event ->
                when (event) {
                    is Request<*> -> corePublisher.request(event)
                    is AdhocEvent -> corePublisher.enqueue(event)
                }
            }
```

or we can use `run` which will still use `this` as the event
```suggestion
            it.event.run {
                when (this) {
                    is Request<*> -> corePublisher.request(this)
                    is AdhocEvent -> corePublisher.enqueue(this)
                }
            }
```

Reply:
In order to avoid conflicts from merge, I will address stylistic changes on next set of PRs. I will link the issue here when I create it

Reply:
https://github.com/ably/ably-asset-tracking-android/issues/556